internal/client/fetcher.go

```diff
@@ -149,7 +151,23 @@ func (h HTTPFetcher) ReadIssuer(ctx context.Context, hash []byte) ([]byte, error
 
 // FileFetcher knows how to fetch log artifacts from a filesystem rooted at Root.
 type FileFetcher struct {
-	Root string
+	Root                    string
+	IsEntryBundleCompressed bool
+}
+
+// decompress decompresses gzip data if needed
+func (f FileFetcher) decompress(data []byte, compressed bool) ([]byte, error) {
+	if !compressed {
+		return data, nil
+	}
```

Comment:
To avoid confusion, I think I'd be tempted to perform this check inside `ReadEntryBundle`:

```go
data, err := os.ReadFile(...)
// ...
 
if f.IsEntryBundleCompressed {
  return decompress(data)
}
return data, nil
```

Reply:
Done.